Fix job scheduled without checking precedence constraints

jmigual opened the pull request:
Good afternoon,

As discussed by e-mail, there was a condition where jobs with precedence constraints could be scheduled even if their precedence constraints were not finished.

This PR fixes that issue. An example that illustrates this issue can be found 
here: [schedule_results.tar.gz](https://github.com/brandenburg/np-schedulability-analysis/files/4461571/schedule_results.tar.gz). There are two directories:

- **before**: Results before applying this change
- **after**: Results after applying this change

The task set has a lot of jobs but the important part is that task 4 depends on task 3 which depends on task 2. The example does not have jitter and thus it should be deterministic so only one state should follow the previous one. In the "before" results we can see that state S17 has two successors one of them consists in scheduling T4J1 which shouldn't be possible as the other option schedules T2J1 (and we have the T4 -> T3 -> T2 dependency). 

After applying this fix, this does not happen anymore and we obtain the results in the "after" directory.

jmigual committed:
Fix job scheduled without checking precedence constraints

A job could be ready because it has been released but it
still has to wait for its precedence constraints jobs to
finish in order to be "totally ready"

brandenburg commented:
Thanks for spotting this. Could you please update the patch to follow the coding style in the rest of the project (`if (cond)` should be on a line by itself)?

brandenburg commented:
Also, ideally, would you mind adding a minimal example exercising the (fix of) the bug as a unit test? That way we avoid re-introducing it later as a regression.

jmigual commented:
Sure, I will try to update it by the end of this week

jmigual committed:
Follow proper code style

Commit:
Add test for precedence inversion case

jmigual commented:
Alright, I've changed the code to follow the coding style and I've also added a test checking for this specific case. The graph output of the tool before and after applying this pull request can be seen here: 
[test_results.tar.gz](https://github.com/brandenburg/np-schedulability-analysis/files/4506531/test_results.tar.gz)

brandenburg commented:
Thanks a lot!
